contentcuration/contentcuration/viewsets/base.py

```diff
@@ -301,22 +301,25 @@ def update(self, queryset, all_validated_data):
             obj_id = self.child.id_value_lookup(obj)
             obj_validated_data = all_validated_data_by_id.get(obj_id)
 
-            # Reset the child serializer changes attribute
-            self.child.changes = []
-            # use model serializer to actually update the model
-            # in case that method is overwritten
-
-            instance = self.child.update(obj, obj_validated_data)
-            # If the update method does not return an instance for some reason
-            # do not try to run further updates on the model, as there is no
-            # object to udpate.
-            if instance:
-                m2m_fields_by_id[obj_id] = self.child.m2m_fields
-                updated_objects.append(instance)
-            # Collect any registered changes from this run of the loop
-            self.changes.extend(self.child.changes)
-
-            updated_keys.add(obj_id)
+            # If no valid data was passed back then this will be None
+            if obj_validated_data is not None:
```

Comment:
@rtibbles Do we know why `all_validated_data_by_id` is missing ids that are in `objects_to_update`? Or is `id_value_lookup` possibly failing?

Reply:
Yes, it's because of updates that do not pass through validation - fields that are not on the serializer for writing.

Two ways this can happen:

1. prerequisite_ids - because they are an M2M field they are handled separately, so don't get passed through here (I think we should eventually remove them from here and not process them in the content node endpoint, but that's separate).

2. complete - this hadn't been added to the serializer as a writeable field, so changes to this field were getting discarded, and no data was present.

In the lines preceding this, we only add data to the `all_validated_data_by_id` if the data is non-empty, so these empty updates are not being added (which is good, because we don't need to update them then!)